### ISSUE TITLE
Type definition?

### ISSUE THREAD

**Issue by justinyoo** [imcvampire/vue-axios#4]
Hi,

Thanks for this library. I really appreciate it. Just wonder if you can provide us with type definition, please?

Cheers,

**Comment by imcvampire** [comment]
I'm sorry, I can't catch your idea about type definition. Do you mean using somethings like Flow???

**Comment by justinyoo** [comment]
I mean, like `vue-axios.d.ts`. I tried to generate a `.d.ts` file by myself using [dts-gen](https://github.com/Microsoft/dts-gen), but didn't work.

**Comment by imcvampire** [comment]
Right now, I don't have a good knowledge about TypeScript! But I will try my best to deliver it ASAP.

And would you mind helping me provide it?

**Comment by johnathankent** [comment]
@justinyoo, @imcvampire
Axios already provides it: https://github.com/mzabriskie/axios#typescript

**Comment by s-KaiNet** [comment]
As quick workaround you can do  
```
npm install typings -g
typings install dt~axios --save --global
```
then in your folder with sources create file `vue-axios-shim.d.ts` and put below content inside:   
```javascript
import Vue from "vue";

declare module "vue/types/vue" {
  interface Vue {
    $http: Axios.AxiosInstance;
    $axios: Axios.AxiosInstance;
  }
}
```

**Comment by asulta** [comment]
Typings is not necessary here.
Axios already provides typescript definitions in its bundle
As @s-KaiNet said, you can write your own definition:

```typescript
import {AxiosInstance} from "axios";

declare module "vue/types/vue" {

    interface Vue {
        axios: AxiosInstance;
        $http: AxiosInstance;
    }

    namespace Vue {
        const axios: AxiosInstance;
    }

}
```
I edited the definition to match the [docs](https://github.com/imcvampire/vue-axios#usage) :)

**Comment by s-KaiNet** [comment]
Right, actually `typings install` is not needed anymore, because TS compiler now reads `package.json` and includes type definitions automatically. I've updated answer if anybody will refer.

**Comment by imcvampire** [comment]
Thank @asulta and @s-KaiNet!!!

Can you create a new pull request update README and/or add some code provide typing if necessary!

**Comment by asulta** [comment]
@s-KaiNet I can make a PR to add ts definitions, but I think that the only way to do that is to create a dev dependency to axios. Is that okay for you?

**Comment by imcvampire** [comment]
@asulta I think we can add `axios` to `dependency` because this package can't work correctly without `axios`

**Comment by s-KaiNet** [comment]
Hey guys, since PR was merged, can you please update the library to include definitions in npm?

**Comment by imcvampire** [comment]
Hi @s-KaiNet, I deployed a new version (`2.1.0`)

**Comment by olegshtch** [comment]
Will there some example with `vue-class-component`?

**Comment by asulta** [comment]
I think that `vue-class-component` does not have effect on how we use `vue-axios`.

4 example:
```typescript
const template = require("./someTemplate.vue");

@Component({ template: template })
export class SomeComponent extends Vue {

  model: any;

  created() {
    this.$http.get('someEndpoint')
      .then(response => this.model = response.data);
    /* 
       or
       this.axios.get('someEndpoint')
         .then(response => this.model = response.data);
    */
  }
```

I have not used `vue-class-component` for quite some time though, so I might be wrong.

**Comment by Nomenator** [comment]
@imcvampire Strangely enough, when updating the package through npm, it gets the new package.json, but not the typescript definitions. Has to be shimmed in manually still (working in Visual Studio 2017, npm from command line).

**Comment by s-KaiNet** [comment]
Yep, the same for me

**Comment by imcvampire** [comment]
@s-KaiNet @s-KaiNet Sorry for the bug. I have published `2.1.1` to address it
